core/src/blocks/extrinsics.rs

```diff
@@ -0,0 +1,717 @@
+        const PALLET: &'static str = "Test";
+        const CALL: &'static str = "TestCall";
+    }
+
+    /// Build fake metadata consisting the types needed to represent an extrinsic.
+    fn metadata() -> Metadata {
+        let pallets = vec![PalletMetadata {
+            name: "Test",
+            storage: None,
+            calls: Some(PalletCallMetadata {
+                ty: meta_type::<Pallet>(),
+            }),
+            event: None,
+            constants: vec![],
+            error: None,
+            index: 0,
+            docs: vec![],
+        }];
+
+        let extrinsic = ExtrinsicMetadata {
+            version: 4,
+            signed_extensions: vec![],
+            address_ty: meta_type::<()>(),
+            call_ty: meta_type::<RuntimeCall>(),
+            signature_ty: meta_type::<()>(),
+            extra_ty: meta_type::<()>(),
+        };
+
+        let meta = RuntimeMetadataV15::new(
+            pallets,
+            extrinsic,
+            meta_type::<()>(),
+            vec![],
+            OuterEnums {
+                call_enum_ty: meta_type::<RuntimeCall>(),
+                event_enum_ty: meta_type::<()>(),
+                error_enum_ty: meta_type::<()>(),
+            },
+            CustomMetadata {
+                map: Default::default(),
+            },
+        );
+        let runtime_metadata: RuntimeMetadataPrefixed = meta.into();
+        let metadata: subxt_metadata::Metadata = runtime_metadata.try_into().unwrap();
+
+        Metadata::from(metadata)
+    }
+
+    #[test]
+    fn extrinsic_metadata_consistency() {
+        let metadata = metadata();
+
+        // Except our metadata to contain the registered types.
+        let pallet = metadata.pallet_by_index(0).expect("pallet exists");
+        let extrinsic = pallet
+            .call_variant_by_index(2)
+            .expect("metadata contains the RuntimeCall enum with this pallet");
+
+        assert_eq!(pallet.name(), "Test");
+        assert_eq!(&extrinsic.name, "TestCall");
+    }
+
+    #[test]
+    fn insufficient_extrinsic_bytes() {
+        let metadata = metadata();
+        let ids = ExtrinsicPartTypeIds::new(&metadata).unwrap();
+
+        // Decode with empty bytes.
+        let result = ExtrinsicDetails::<SubstrateConfig>::decode_from(0, &[], metadata, ids);
+        assert_matches!(result.err(), Some(crate::Error::Codec(_)));
+    }
+
+    #[test]
+    fn unsupported_version_extrinsic() {
+        let metadata = metadata();
+        let ids = ExtrinsicPartTypeIds::new(&metadata).unwrap();
+
+        // Decode with invalid version.
+        let result =
+            ExtrinsicDetails::<SubstrateConfig>::decode_from(0, &vec![3u8].encode(), metadata, ids);
+
+        assert_matches!(
+            result.err(),
+            Some(crate::Error::Block(
+                crate::error::BlockError::UnsupportedVersion(3)
+            ))
+        );
+    }
+
+    // TODO add this back when we have tx encode logic in core.
```

Comment:
Can you open an issue for this as well? Easy to forgot

What is stopping us to have tx encode logic in core?

Reply:
Ooh good point; I added this TODO when I hadn't ported over the extrinsic creation stuff to subxt-core, but now I think I can re-enable that test or something :)

Reply:
Re-enabled it!